git_prompt.lua

```diff
@@ -6,6 +6,9 @@ local gitutil = require('gitutil')
 -- this code is stolen from https://github.com/Dynodzzo/Lua_INI_Parser/blob/master/LIP.lua
 -- Resolve licensing issues before exposing
 local function load_ini(fileName)
+    -- Check for Cmder configured Git Status Opt In/Out - See: https://github.com/cmderdev/cmder/issues/2484
+    if cmderGitStatusOptIn == false then return nil end
```

Comment:
@daxgames it feels like this can be done on a higher level - you probably can check this at the beginning of `git_prompt_filter` function and return false imeediately so the whole prompt filter is skipped and `load_ini` not called at all.

Reply:
Check for global has been moved

Reply:
Yes, I think the natural place to short circuit would be at the top of `git_prompt_filter`, where other short circuit exits are also located.  Any further out and it could have negative side effects.  Any further in and it could potentially allow other work to occur that might be undesirable in some scenario we haven't identified yet.

Just for completeness, I don't see any suggestions to put a check in `load_ini`:

@istagam observed that `load_ini` is where the expensive work occurs and suggested to not execute `git-prompt.lua` at all.  That's impractical because a user might `cd` to different directories, and might want a prompt in some and not in others.

I didn't make any specific suggestion, I only said "Cmder's git prompt can set it each time saying whether it skipped filtering the prompt, and clink-completions can check the variable."  I didn't have time available to identify specifically where to put it, so it seemed reasonable to defer that assessment rather than risk making a suggestion that missed some important details.